com.unity.multiplayer.mlapi/Runtime/Core/NetworkedBehaviour.cs

```diff
@@ -308,122 +308,197 @@ internal void InitializeVars()
+                        {
+                            sobj.childNetworkedBehaviours[k].PostNetworkVariableWrite();
+                        }
+                    }
+                }
+                else
+                {
+
+                    // when client updates the sever, it tells it about all its objects
+                    foreach (var sobj in SpawnManager.SpawnedObjectsList)
+                    {
+                        for (int k = 0; k < sobj.childNetworkedBehaviours.Count; k++)
+                        {
+                           sobj.childNetworkedBehaviours[k].VarUpdate(NetworkingManager.Singleton.ServerClientId);
+                        }
+                    }
+
+                    // Now, reset all the no-longer-dirty variables
+                    // again, should keep a list of objects I touched
+                    foreach (var sobj in SpawnManager.SpawnedObjectsList)
+                    {
+                        for (int k = 0; k < sobj.childNetworkedBehaviours.Count; k++)
+                        {
+                            sobj.childNetworkedBehaviours[k].PostNetworkVariableWrite();
+                        }
+                    }
+                }
+
+            }
+            finally
+            {
+#if DEVELOPMENT_BUILD || UNITY_EDITOR
+                s_NetworkedBehaviourUpdate.End();
+#endif
+            }
+        }
+
+
+        internal void PreNetworkVariableWrite()
+        {
+            // reset our "which variables got written" data
+            networkedVarIndexesToReset.Clear();
+            networkedVarIndexesToResetSet.Clear();
+        }
+
+        internal void PostNetworkVariableWrite()
+        {
+            // mark any variables we wrote as no longer dirty
+            for (int i = 0; i < networkedVarIndexesToReset.Count; i++)
+            {
+                networkedVarFields[networkedVarIndexesToReset[i]].ResetDirty();
+            }
+        }
+
+        internal void VarUpdate(ulong clientId)
         {
             if (!varInit)
                 InitializeVars();
 
-            NetworkedVarUpdate();
+            PreNetworkVariableWrite();
+            NetworkedVarUpdate(clientId);
         }
 
         private readonly List<int> networkedVarIndexesToReset = new List<int>();
         private readonly HashSet<int> networkedVarIndexesToResetSet = new HashSet<int>();
 
-        private void NetworkedVarUpdate()
+        private void NetworkedVarUpdate(ulong clientId)
         {
-            if (!CouldHaveDirtyNetworkedVars())
-                return;
+           if (!CouldHaveDirtyNetworkedVars())
+               return;
 
-            networkedVarIndexesToReset.Clear();
-            networkedVarIndexesToResetSet.Clear();
-
-            for (int i = 0; i < (IsServer ? NetworkingManager.Singleton.ConnectedClientsList.Count : 1); i++)
+            for (int j = 0; j < channelMappedNetworkedVarIndexes.Count; j++)
```

Comment:
Git is making a big mess of this diff.  This isn't as huge of a change as it looks.   This 'j' loop appears on line 336 on the other side.